src/sqlfluff/rules/L042.py

```diff
@@ -48,44 +70,281 @@ class Rule_L042(BaseRule):
+    Seg = partial(_get_seg, dialect=dialect)
+    ctes = _CTEChecker()
+    is_with = root_select.all(is_type("with_compound_statement"))
+    # TODO: consider if we can fix recursive CTEs
+    is_recursive = is_with and len(root_select.children(is_name("recursive"))) > 0
+    segmentify = partial(
+        _segmentify,
+        casing=_get_casing_preference(root_select),
+    )
+    # Init the output/final select &
+    # populate existing CTEs
+    output_select = root_select
+    if is_with:
+        output_select = root_select.children(
+            is_type(
+                "set_expression",
+                "select_statement",
+            )
+        )
+        for cte in root_select.children(is_type("common_table_expression")):
+            assert isinstance(cte, CTEDefinitionSegment), "TypeGaurd"
+            ctes.insert_cte(cte)
+
+    lint_results: List[LintResult] = []
+    CTESegment = Seg(CTEDefinitionSegment)
+    TableExpressionSeg = Seg(TableExpressionSegment)
+    TableReferenceSeg = Seg(TableReferenceSegment)
+    for parent_type, _, this_seg, subquery in nested_subqueries:
+        alias_name = ctes.get_name(this_seg.children(is_type("alias_expression")))
+        new_cte = CTESegment(
+            segments=(
+                CodeSegment(
+                    raw=alias_name,
+                    name="naked_identifier",
+                    type="identifier",
+                ),
+                WhitespaceSegment(),
+                segmentify("AS"),
+                WhitespaceSegment(),
+                subquery,
+            )
+        )
+        # (mypy is not understanding the generic)
+        ctes.insert_cte(new_cte)  # type: ignore
+        # TODO: Create non-mutative helper function.
+        # We will replace the whole tree, mutate the table expression.
+        this_seg[0].segments = (
+            TableExpressionSeg(
+                segments=(
+                    TableReferenceSeg(
+                        segments=(
+                            CodeSegment(
+                                raw=alias_name,
+                                name="naked_identifier",
+                                type="identifier",
+                            ),
+                        )
+                    ),
                 )
-                if seg:
-                    return LintResult(
-                        anchor=seg[0],
-                        description=f"{parent_type} clauses should not contain "
-                        "subqueries. Use CTEs instead",
-                    )
-        return None
+            ),
+        )
+        res = LintResult(
+            anchor=subquery,
+            description=f"{parent_type} clauses should not contain "
+            "subqueries. Use CTEs instead",
+            fixes=[],
+        )
+        lint_results.append(res)
+
+    if ctes.has_duplicates() or is_recursive:
+        # If we have duplicate CTE names just don't fix anything
```

Comment:
Should we return a `LintResult` here to warn about the issue (not try to fix it, though)?

Reply:
We do line 194 will have a fixless Lint Result